### DISCUSSION TITLE
configure the direction in which new displays are positioned

### DISCUSSION THREAD

**Discussion by MangoIV** [YaLTeR/niri#610] [Ideas]
Hi ~ first of all, many thanks for this great compositor, it is really nice to use. 

So the situation is as follows: I often dock my laptop on my monitor and I use the same niri config across multiple laptops. I would like the monitor to always be on top of the laptop. This is not trivial because absolute sizes differ wildly between the laptop monitors. So it would be really nice to be able to configure something like `output.positioning` with the options:
- `{bottom,top}-{left,right,center}
- `{left, right}-{bottom,top,centre}`

This way one wouldn't have to think of output specific logic and could just let niri set it automatically.

**Comment by YaLTeR** [comment]
Hey! The problem is that this would rely on the internal output order (i.e. what if you have two outputs with "top", which one goes topmost and which one second?). Currently niri sorts outputs by connector name, though in the future it'll probably sort them by monitor manufacturer and model.

If you have two outputs that you want to position this way and they happen to sort in a different order than you need, you're out of luck. Ideally the system would not rely on this sort order.

Btw, you should be able to use [kanshi](https://sr.ht/~emersion/kanshi/) to reconfigure the monitors, perhaps currently manually since only connector names are available, but in the future with manufacturer and model it should work automatically.

**Reply by MangoIV** [reply]
hmmm, I don't know what the general solution there but maybe it should be possible to set a "main" monitor and everything else will be stacked from there on in arbitrary order. I mean it doesn't really solve the issue for everyone, but I think having a large number of monitors is more of an edge case vs. setting your laptop left, right or under your screen seems to come up often... 

but you're right, all of this isn't ideal...

Best would be if `wdisplays` would work but somehow it doesn't, at least  not with fractional scaling...
